lib/src/just_the_tooltip.dart

```diff
@@ -431,19 +431,34 @@ abstract class _JustTheTooltipState<T> extends State<JustTheInterface>
   /// hidden after the current animation completes. In either case,
   /// this future will complete once the tooltip has been completely hidden.
   Future<void> _hideTooltip({bool immediately = false}) async {
+    if (_animationController.isDismissed ||
+        _animationController.status == AnimationStatus.reverse) {
+      // Tooltip is already hiding or animating out, do nothing.
+      return;
+    }
     cancelShowTimer();
+    final AnimationController proxy = AnimationController(
+      value: _animationController.value,
+      vsync: this,
+    );
+    void updateProxy() {
+      proxy.value = _animationController.value;
+    }
 
-    final completer = Completer<void>();
-    final future = completer.future;
-
-    if (immediately) {
-      widget.onDismiss?.call();
+    _animationController.addListener(updateProxy);
+    widget.onDismiss?.call(proxy);
 
+    final completer = Completer<void>();
+    final future = completer.future.whenComplete(() {
+      _removeEntries();
```

Comment:
There was some really weird behavior around the completers in `_hideTooltip` and `showTooltip`. Basically, hide and show were only kicking off the animation when they were first called and then setting the controller's status when the animation finished. Then, a listener on the animation status would see that it had changed and would call `_hide` or `_show` respectively a second time, this time with `immediately = true`. The functions would then properly execute clean up logic under an `if (immediately)` check. This was incorrectly firing the `onShow` and `onDismiss` callbacks twice.

I've fixed this by putting all the clean up logic in `completer.complete()` so it's called by the immediate and non-immediate code paths and by adding a check at the top of each function to short circuit it if the tooltip is already in the desired state.